cmd/helm/main.go

```diff
@@ -0,0 +1,226 @@
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "helm.core.rukpak.io",
+		NewCache: cache.BuilderWithOptions(cache.Options{
+			SelectorsByObject: cache.SelectorsByObject{
+				&rukpakv1alpha1.BundleDeployment{}: {},
+				&rukpakv1alpha1.Bundle{}:           {},
+			},
+			DefaultSelector: cache.ObjectSelector{
+				Label: dependentSelector,
+			},
+		}),
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to create manager")
+		os.Exit(1)
+	}
+
+	ns := util.PodNamespace(systemNamespace)
+	storageURL, err := url.Parse(fmt.Sprintf("%s/bundles/", httpExternalAddr))
+	if err != nil {
+		setupLog.Error(err, "unable to parse bundle content server URL")
+		os.Exit(1)
+	}
+
+	localStorage := &storage.LocalDirectory{
+		RootDirectory: storageDirectory,
+		URL:           *storageURL,
+	}
+
+	var rootCAs *x509.CertPool
+	if bundleCAFile != "" {
+		var err error
+		if rootCAs, err = util.LoadCertPool(bundleCAFile); err != nil {
+			setupLog.Error(err, "unable to parse bundle certificate authority file")
+			os.Exit(1)
+		}
+	}
+
+	httpLoader := storage.NewHTTP(
+		storage.WithRootCAs(rootCAs),
+		storage.WithBearerToken(cfg.BearerToken),
+	)
+	bundleStorage := storage.WithFallbackLoader(localStorage, httpLoader)
+
+	// NOTE: AddMetricsExtraHandler isn't actually metrics-specific. We can run
+	// whatever handlers we want on the existing webserver that
+	// controller-runtime runs when MetricsBindAddress is configured on the
+	// manager.
+	if err := mgr.AddMetricsExtraHandler("/bundles/", bundleStorage); err != nil {
+		setupLog.Error(err, "unable to register bundle content server")
+		os.Exit(1)
+	}
+
+	// This finalizer logic MUST be co-located with this main
+	// controller logic because it deals with cleaning up bundle data
+	// from the bundle cache when the bundles are deleted. The
+	// consequence is that this process MUST remain running in order
+	// to process DELETE events for bundles that include this finalizer.
+	// If this process is NOT running, deletion of such bundles will
+	// hang until $something removes the finalizer.
+	//
+	// If the bundle cache is backed by a storage implementation that allows
+	// multiple writers from different processes (e.g. a ReadWriteMany volume or
+	// an S3 bucket), we could have separate processes for finalizer handling
+	// and the primary helm provisioner controller. For now, the assumption is
+	// that we are not using such an implementation.
+	bundleFinalizers := crfinalizer.NewFinalizers()
+	if err := bundleFinalizers.Register(finalizer.DeleteCachedBundleKey, &finalizer.DeleteCachedBundle{Storage: bundleStorage}); err != nil {
+		setupLog.Error(err, "unable to register finalizer", "finalizerKey", finalizer.DeleteCachedBundleKey)
+		os.Exit(1)
+	}
+
+	unpacker := source.NewUnpacker(map[rukpakv1alpha1.SourceType]source.Unpacker{
+		rukpakv1alpha1.SourceTypeHTTP: &source.HTTP{
+			Reader:          mgr.GetAPIReader(),
+			SecretNamespace: systemNamespace,
+		},
+	})
```

Comment:
Any reason we shouldn't support all of the source types in the helm provisioner?

Reply:
Oh, I didn't think about it.  There is no reason that we cannot support the other sources.

Reply:
Does the registry provisioner support the gitsource?   In theory, it can but have we tested?